### PR TITLE
Add DeadChannelMap to GenericCheck and improve EventsInGateTime histogram

### PR DESCRIPTION
**Pull request by DawidSkora**
Add DeadChannelMap to GenericCheck and use in MaxThresholdY and MinThresholdY.
Improve EventsInGateTime histogram by adding axis labels, better title and improvement for MaxThresholdY.

### CODE REVIEW

File: Modules/FIT/FV0/src/GenericCheck.cxx [hunk 1]
```diff
@@ -82,6 +82,47 @@ void GenericCheck::configure()
...
+  }
+
+  // Print DeadChannelMap
+  mDeadChannelMapStr = "";
+  for (unsigned chId = 0; chId < mDeadChannelMap->map.size(); chId++) {
+    if (!mDeadChannelMap->isChannelAlive(chId)) {
+      mDeadChannelMapStr += (mDeadChannelMapStr.empty() ? "" : ",") + std::to_string(chId);
+    }
+  }
+  if (mDeadChannelMapStr.empty())
```
Comment (Barthelemy) [inline Modules/FIT/FV0/src/GenericCheck.cxx:122]: The coding guidelines state that curvy brackets must be used even if there is a single line in the body of the `if` or `for`.

File: Modules/FIT/FV0/src/GenericCheck.cxx [hunk 2]
```diff
@@ -158,9 +199,10 @@ Quality GenericCheck::check(std::map<std::string, std::shared_ptr<MonitorObject>
       }
 
       if (mCheckMinThresholdY.isActive()) {
-        int numberOfBinsX = h->GetNbinsX();
         float minValue = h->GetBinContent(1);
-        for (int channel = 1; channel < numberOfBinsX; ++channel) {
+        for (int channel = 1; channel < h->GetNbinsX(); ++channel) {
+          if (channel >= sNCHANNELS || !mDeadChannelMap->isChannelAlive(channel))
```
Comment (Barthelemy) [inline Modules/FIT/FV0/src/GenericCheck.cxx:204]: same

File: Modules/FIT/FV0/src/GenericCheck.cxx [hunk 3]
```diff
@@ -170,9 +212,21 @@ Quality GenericCheck::check(std::map<std::string, std::shared_ptr<MonitorObject>
...
-        mCheckMaxThresholdY.mBinNumberX = h->GetMaximumBin();
-        float maxValue = h->GetBinContent(mCheckMaxThresholdY.mBinNumberX);
-        mCheckMaxThresholdY.doCheck(result, maxValue);
+        if (mDeadChannelMap->isChannelAlive(h->GetMaximumBin())) {
+          mCheckMaxThresholdY.mBinNumberX = h->GetMaximumBin();
+          mCheckMaxThresholdY.doCheck(result, h->GetBinContent(mCheckMaxThresholdY.mBinNumberX));
+        } else {
+          float maxValue = 0;
+          for (int channel = 1; channel < h->GetNbinsX(); ++channel) {
+            if (channel >= sNCHANNELS || !mDeadChannelMap->isChannelAlive(channel))
```
Comment (Barthelemy) [inline Modules/FIT/FV0/src/GenericCheck.cxx:221]: same

**Review by Barthelemy — CHANGES_REQUESTED** [review]
The code is fine , there is just this thing with the missing curvy brackets.

File: Modules/FIT/FV0/src/GenericCheck.cxx (hunk 1 above)
Comment (DawidSkora) [reply Modules/FIT/FV0/src/GenericCheck.cxx]: Done

File: Modules/FIT/FV0/src/GenericCheck.cxx (hunk 2 above)
Comment (DawidSkora) [reply Modules/FIT/FV0/src/GenericCheck.cxx]: Done

File: Modules/FIT/FV0/src/GenericCheck.cxx (hunk 3 above)
Comment (DawidSkora) [reply Modules/FIT/FV0/src/GenericCheck.cxx]: Done

**Comment by jotwinow** [comment]
@Barthelemy @knopers8 could you merge it asap. It is a base for FT0 and FDD to be implemented this week.

**Review by Barthelemy — APPROVED** [review]

**Comment by Barthelemy** [comment]
@jotwinow what should I do ? wait for the fix requested, rightfully, from Andreas ?

File: Modules/FIT/FV0/src/GenericCheck.cxx [hunk 4]
```diff
@@ -82,6 +82,48 @@ void GenericCheck::configure()
...
+    ILOG(Debug, Support) << "configure() : using pathDeadChannelMap: " << mPathDeadChannelMap << ENDM;
+  } else {
+    mPathDeadChannelMap = "FV0/Calib/DeadChannelMap";
+    ILOG(Debug, Support) << "configure() : using default pathDeadChannelMap: " << mPathDeadChannelMap << ENDM;
+  }
+
+  // Align mDeadChannelMap with downloaded one
+  std::map<std::string, std::string> metadata;
+  mDeadChannelMap = retrieveConditionAny<o2::fit::DeadChannelMap>(mPathDeadChannelMap, metadata, (long)-1);
+  if (!mDeadChannelMap->map.size()) {
```
[the remark this answers is not in the document]
Comment (DawidSkora) [reply Modules/FIT/FV0/src/GenericCheck.cxx]: Thanks for your comment, I changed it according to your suggestion.

**Comment by DawidSkora** [comment]
> @jotwinow what should I do ? wait for the fix requested, rightfully, from Andreas ?

I added fix according to Andreas comment.

**Comment by jotwinow** [comment]
@DawidSkora please add the fix suggested by Andreas and add it to PR

**Comment by jotwinow** [comment]
> @jotwinow what should I do ? wait for the fix requested, rightfully, from Andreas ?

@Barthelemy  Please take the fix by David and merge it please.

**Comment by jotwinow** [comment]
@knopers8 @Barthelemy could you approve and merge it

**Pull request merged**